docs/docs/reference/enums/adts.md

```diff
@@ -52,11 +52,15 @@ Note that the type of the expressions above is always `Option`. That
 is, the implementation case classes are not visible in the result
 types of their `apply` methods. This is a subtle difference with
```

Comment:
This should be removed because case classes and enums have the same signatures in the apply method